tests/clickhouse-test

```diff
@@ -360,39 +358,6 @@ def clickhouse_execute_json(
     return rows
 
 
-class Terminated(KeyboardInterrupt):
-    pass
-
-
-def signal_handler(sig, frame):
-    raise Terminated(f"Terminated with {sig} signal")
-
-
-def stop_tests():
-    global stop_tests_triggered_lock
-    global stop_tests_triggered
-    global restarted_tests
-
-    with stop_tests_triggered_lock:
-        print("Stopping tests")
-        if not stop_tests_triggered.is_set():
-            stop_tests_triggered.set()
-
-            # materialize multiprocessing.Manager().list() object before
-            # sending SIGTERM since this object is a proxy, that requires
-            # communicating with manager thread, but after SIGTERM will be
-            # send, this thread will die, and you will get
-            # ConnectionRefusedError error for any access to "restarted_tests"
-            # variable.
-            restarted_tests = [*restarted_tests]
-
-            # send signal to all processes in group to avoid hung check triggering
-            # (to avoid terminating clickhouse-test itself, the signal should be ignored)
-            signal.signal(signal.SIGTERM, signal.SIG_IGN)
-            os.killpg(os.getpgid(os.getpid()), signal.SIGTERM)
-            signal.signal(signal.SIGTERM, signal.SIG_DFL)
```

Comment:
Hm, this part looks useful (see also for some comments https://github.com/ClickHouse/ClickHouse/pull/58213)

In a nutshell this should avoid leaving left overs when the clickhouse-test terminated, since there are bunch of .sh tests that could do tricky things

Reply:
I've returned it. After checking the PR i didn't see any leftovers locally after clickhouse-server or process termination

Reply:
You should run some tricky test, some `.sh` test, to see them (for instance the one that called stress or similar, that has loop in background + wait)